source/Ship.cpp

```diff
@@ -464,8 +464,26 @@ void Ship::Load(const DataNode &node)
 			description += child.Token(1);
 			description += '\n';
 		}
-		else if(key != "actions")
-			child.PrintTrace("Skipping unrecognized attribute:");
+		else if(key != "actions") {
+			std::string actualKey = "";
+
+			if(variantName.length()) {
+				if(key == "-bay")
+					actualKey = key.substr(1);
+			}
+
+			if(actualKey.length()) {
+				if(!partsToRemove)
+					partsToRemove = new std::map<std::string, int>();
+
+				if(!partsToRemove->count(actualKey))
+					partsToRemove->insert(std::pair<std::string, int>(actualKey, 0));
+				partsToRemove->at(actualKey)++;
+
+			}
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
```

Comment:
This needs to be changed back. 
At the top of all these `if`s, you should check if the first token in the node is `remove`, and that it is followed by `bays` (or `guns`, `turrets`, or `weapons`, the exact syntax may change), if it isn't, print a trace indicating unsupported use of the `remove` token.
Near the bottom, above this last `else if`, check if the node is `remove bays` and set the bool created in Ship.h to true.
Probably also set the appropriate bool for weapons and whatever else we are supporting removal of if the relevant tokens are present in the variant definition.